Tests/GeoLib/TestAABB.cpp

```diff
@@ -177,6 +190,11 @@ TEST(GeoLibAABB, RandomNumberOfPointsRandomBox)
 	 ASSERT_LE(minus_half_box_size_x, min_pnt[0]) << "coordinate 0 of min_pnt is smaller than " << minus_half_box_size_x;
 	 ASSERT_LE(minus_half_box_size_y, min_pnt[1]) << "coordinate 1 of min_pnt is smaller than " << minus_half_box_size_y;
 	 ASSERT_LE(minus_half_box_size_z, min_pnt[2]) << "coordinate 2 of min_pnt is smaller than " << minus_half_box_size_z;
+
+	// since the interval is half-open we have to move the upper bound also
+	 half_box_size_x = std::nexttoward(half_box_size_x, std::numeric_limits<double>::max());
+	 half_box_size_y = std::nexttoward(half_box_size_y, std::numeric_limits<double>::max());
+	 half_box_size_z = std::nexttoward(half_box_size_z, std::numeric_limits<double>::max());
```

Comment:
That's why I don't like your solution that fiddles around with 'infinitesimal floats': It requires this ugly fiddlings also in other places. ^^